### DISCUSSION TITLE
What is the point of the Assembly language operations like this: MOV EAX, [ESP+4] if we could just move the stack pointer +4 of the current address?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#134327] [Programming Help]
### Body

What is the point of the Assembly language operations like this: MOV EAX, [ESP+4] if we could just move the stack pointer +4 of the current address?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Buddhi19** [comment]
If you were to adjust ESP directly, it would alter the stack structure,  making it impossible to reliably access other stack data such as return address or other parameters.

**Reply by ghost** [reply]
I just graduated from high school and I need a web developer to help me build the frontend for an open sourced OS I am going to build in Assembly language. Do you know anyone who can help me with either one?
